shared/teams/team/really-remove-member/index.tsx

```diff
@@ -1,35 +1,74 @@
 import * as React from 'react'
 import * as Kb from '../../../common-adapters'
 import * as Styles from '../../../styles'
+import * as RouteTreeGen from '../../../actions/route-tree-gen'
+import * as Container from '../../../util/container'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as TeamsGen from '../../../actions/teams-gen'
 
-type Props = {
-  onClose: () => void
-  onRemove: () => void
-  member: string
-  name: string
-}
+type Props = Container.RouteProps<{
+  username: string
+  teamID: Types.TeamID
+  navToChat?: boolean
+}>
 
-const ReallyRemoveMember = (props: Props) => (
-  <Kb.ConfirmModal
-    confirmText={`Yes, remove ${props.member}`}
-    description={`${props.member} will lose access to all the ${props.name} chats and folders, and they won't be able to get back unless an admin invites them.`}
-    header={
-      <Kb.Box style={styles.iconContainer}>
-        <Kb.Avatar username={props.member} size={64} />
-        <Kb.Box style={styles.icon}>
-          <Kb.Icon color={Styles.globalColors.red} type="iconfont-remove" />
-        </Kb.Box>
-      </Kb.Box>
-    }
-    onCancel={props.onClose}
-    onConfirm={props.onRemove}
-    prompt={
-      <Kb.Text center={true} type={Styles.isMobile ? 'Header' : 'HeaderBig'} style={styles.header}>
-        Are you sure you want to remove {props.member} from&nbsp;{props.name}?
-      </Kb.Text>
+const ReallyRemoveMember = (props: Props) => {
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  const member = Container.getRouteProps(props, 'username', '')
+  const navToChat = Container.getRouteProps(props, 'navToChat', false)
+
+  const teamname = Container.useSelector(state => Constants.getTeamNameFromID(state, teamID))
+  const waitingKey = Constants.removeMemberWaitingKey(teamID, member)
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+
+  const onRemove = () =>
+    dispatch(
+      TeamsGen.createRemoveMember({
+        teamID,
+        username: member,
+      })
+    )
+
+  const onClose = () => dispatch(nav.safeNavigateUpPayload())
+
+  const waiting = Container.useAnyWaiting(waitingKey)
+  const wasWaiting = Container.usePrevious(waiting)
+  React.useEffect(() => {
+    if (wasWaiting && !waiting) {
+      if (navToChat) {
+        dispatch(nav.safeNavigateUpPayload())
+      } else {
+        dispatch(RouteTreeGen.createNavUpToScreen({routeName: 'team'}))
+      }
     }
-  />
-)
+  }, [dispatch, waiting, wasWaiting])
```

Comment:
`dispatch` never changes, so I think this could just depend on `[dispatch, waiting]` and check `waiting`'s value inside. But this is more explicit and I'm OK with it too.

Reply:
I'm confused what you mean by the second half of the first sentence. Re dispatch never changing, the linter doesn't know that so it's easier just to include it 🤷

Reply:
Sorry for the confusion; what I meant was, this could simply be:

```
  const waiting = Container.useAnyWaiting(waitingKey)
  React.useEffect(() => {
    if (!waiting) {
      if (navToChat) {
        dispatch(nav.safeNavigateUpPayload())
      } else {
        dispatch(RouteTreeGen.createNavUpToScreen({routeName: 'team'}))
      }
    }
  }, [dispatch, waiting])
```
Because `dispatch` never changes, the only thing driving this effect is `waiting`. So there's no need to track the previous state.

Reply:
Wouldn't that dispatch the action immediately upon mount (since waiting starts false, then becomes true then false again)?

Reply:
Oooh that's right! Nvm then. Thanks